### PR TITLE
Feature 50 add update function for building

### PR DESCRIPTION
**Pull request by JoseBlois**

### CODE REVIEW

File: src/controllers/buildingsController.js [hunk 1]
```diff
@@ -63,9 +63,42 @@ const deleteBuilding = async (req, res) => {
...
+  }
+  const update = {
+    name: req.body.name,
+    phone: req.body.phone,
+    address: req.body.address,
+    company: req.body.company,
+    boilers: req.body.boilers,
+  };
+  try {
+    const result = await models.Building.findByIdAndUpdate(id, update);
```
Comment (lmalvasia) [inline src/controllers/buildingsController.js:86]: >
```suggestion
    const result = await ClientSchema.findByIdAndUpdate(id, { ...req.body }, { new: true });
```

File: src/controllers/buildingsController.js [hunk 2]
```diff
@@ -63,9 +63,42 @@ const deleteBuilding = async (req, res) => {
...
+    return res.status(500).json({
+      msg: 'Missing data'
+    });
+  }
+  if (!req.body.name || !req.body.phone || !req.body.address || !req.body.boilers) {
+    return res.status(500).json({
+      msg: 'Missing required fields to create a building'
+    });
+  }
+  const update = {
```
Comment (lmalvasia) [inline src/controllers/buildingsController.js:78]: You can remove this const and pass the body directly to the findByIdAndUpdate function.

File: src/controllers/buildingsController.js [hunk 3]
```diff
@@ -63,9 +63,42 @@ const deleteBuilding = async (req, res) => {
   }
 };
 
+const updateBuilding = async (req, res) => {
+  const { id } = req.params;
+  if (!req.body) {
```
Comment (lmalvasia) [inline src/controllers/buildingsController.js:68]: Is this validation required? Why?

File: src/controllers/buildingsController.js [hunk 4]
```diff
@@ -63,9 +63,42 @@ const deleteBuilding = async (req, res) => {
...
+    phone: req.body.phone,
+    address: req.body.address,
+    company: req.body.company,
+    boilers: req.body.boilers,
+  };
+  try {
+    const result = await models.Building.findByIdAndUpdate(id, update);
+    return res.status(200).json({
+      msg: 'Object updated succesfully',
+      result
```
Comment (lmalvasia) [inline src/controllers/buildingsController.js:89]: The result is the updated building or the previous building?
If you don't set the option `{ new: true }`, I think the result will be the previous building (not the updated building). Could you check this?

**Review by lmalvasia — CHANGES_REQUESTED** [review]

File: src/controllers/buildingsController.js (hunk 3 above)
Comment (JoseBlois) [reply src/controllers/buildingsController.js:68]: I didnt check to company field down below i thought it was necessary

**Review by lmalvasia — APPROVED** [review]

**Pull request merged**